### PR TITLE
`v0.6.0`

### PR DESCRIPTION
**Pull request by mrkaye97**
* Rips out lots of unused code that's only useful for MLFlow running on local (which is not the goal of this project to enable)
* Rips out the MLFlow `models` infrastructure. The idea for this package -- at least in the short-term -- is to save models as artifacts with the intention of using them in Plumber APIs, etc. If you wanted to load an (e.g.) XGBoost model into Sagemaker, you could use `log_artifact` in tandem with `FUN = xgboost::xgb.save` to make this easy.
* Adds some more helpful hints for MLFlow clients

@tonyelhabr I'm thinking about submitting this one to CRAN

### CODE REVIEW

File: R/globals.R [hunk 1]
```diff
@@ -79,12 +71,7 @@ set_tracking_uri <- function(uri) {
...
 #'
-#' @importFrom fs path_abs
-#'
 #' @export
 get_tracking_uri <- function() {
-  .globals$tracking_uri %||% {
-    env_uri <- Sys.getenv("MLFLOW_TRACKING_URI")
-    if (nchar(env_uri)) env_uri else paste("file://", path_abs("mlruns"), sep = "")
-  }
+  .globals$tracking_uri %||% Sys.getenv("MLFLOW_TRACKING_URI")
```
Comment (tonyelhabr) [inline R/globals.R:76]: might be worth pointing out in the Roxygen that `MLFLOW_TRACKING_URI` can be used like you do in `create_experiment`

File: R/utils.R [hunk 2]
```diff
@@ -1,7 +1,3 @@
-mlflow_relative_paths <- function(paths) {
-  gsub(paste0("^", file.path(getwd(), "")), "", paths)
-}
-
```
Comment (tonyelhabr) [inline R/utils.R:4]: lol this is hacky af so i'm glad we can get rid of this

File: R/observer.R [hunk 3]
```diff
@@ -1,35 +0,0 @@
...
-#' event such as "create_run", "delete_run", or "log_metric".
-#' Each observer should have a `register_tracking_event(event_name, data)`
-#' callback accepting a character vector `event_name` specifying the name of
-#' the tracking event, and `data` containing a list of attributes of the event.
-#' The callback should be non-blocking, and ideally should complete
-#'  instantaneously. Any exception thrown from the callback will be ignored.
-#'
-#' @param observer The observer object (see example)
-#' @export
-register_external_observer <- function(observer) {
```
Comment (tonyelhabr) [inline R/observer.R:14]: i'm fine with getting rid of this. this seems very niche and is sort of against the opinionated nature of the package

File: R/model-crate.R [hunk 4]
```diff
@@ -1,46 +0,0 @@
-#' @rdname save_model
-#' @export
-save_model.crate <- function(model, path, model_spec = list(), ...) {
```
Comment (tonyelhabr) [inline R/model-crate.R:3]: i do like being able to get rid of this, in favor of using `FUN` directly where appropriate

**Review by tonyelhabr — APPROVED** [review]
lgtm. def make sure that the tests pass, which i'm sure you did

File: R/model-crate.R (hunk 4 above)
Comment (mrkaye97) [reply R/model-crate.R:3]: Yeah, I really don't get the point of this at all. It seems like they're just extending the python functionality, but if you're going to load a literal R function, why not just load the model?

File: R/observer.R (hunk 3 above)
Comment (mrkaye97) [reply R/observer.R:14]: Yeah I have no idea what this even does

File: R/globals.R (hunk 1 above)
Comment (mrkaye97) [reply R/globals.R:76]: yeah fair point, will do

**Pull request merged**